### PR TITLE
optional force flag added

### PR DESCRIPTION
**Pull request by tauhid621**
Optional force flag added. If force flag is on and patch fails, then resource will be deleted and re-created.
#24

### CODE REVIEW

File: lib/actions/promote.js [hunk 1]
```diff
@@ -1,51 +1,52 @@
...
-        try {
-            canaryDeploymentHelper.deleteCanaryDeployment(kubectl, TaskInputParameters.manifests, includeServices);
-        }
-        catch (ex) {
-            core.warning('Exception occurred while deleting canary and baseline workloads. Exception: ' + ex);
-        }
-    });
-}
-exports.promote = promote;
+'use strict';
```
Comment (ds-ms) [inline lib/actions/promote.js:1]: @tauhid621 could you undo your LF to CRLF changes

File: src/run.ts [hunk 2]
```diff
@@ -61,10 +61,12 @@ export async function run() {
     let action = core.getInput('action');
     let manifests = manifestsInput.split('\n');
 
+    let forceDeployment: boolean = core.getInput('force') === 'true';
+
     if (action === 'deploy') {
         let strategy = core.getInput('strategy');
         console.log("strategy: ", strategy)
-        await deploy(new Kubectl(kubectlPath, namespace), manifests, strategy);
+        await deploy(new Kubectl(kubectlPath, namespace, undefined, forceDeployment), manifests, strategy);
```
Comment (ds-ms) [inline src/run.ts:69]: Send `false` instead of `undefined`

Comment (ds-ms) [reply src/run.ts:69]: Is it being set to `true` anywhere btw?

**Comment by shigupt202** [comment]
Can you update the tests for the changes that you've made? The CI checks are failing.

File: src/kubectl-object-model.ts [hunk 3]
```diff
@@ -21,7 +23,13 @@ export class Kubectl {
...
 
     public apply(configurationPaths: string | string[]) {
-        return this.execute(['apply', '-f', this.createInlineArray(configurationPaths)]);
+        if (this.forceDeployment) {
+            console.log("force flag is on, deployment will continue even if previous deployment already exists");
+            return this.execute(['apply', '--force', '-f', this.createInlineArray(configurationPaths)]);
+        }
+        else {
+            return this.execute(['apply', '-f', this.createInlineArray(configurationPaths)]);
+        }
```
Comment (ds-ms) [inline src/kubectl-object-model.ts:32]: Call execute from one place.

Do something like
```
initiate array
if(force) add --force to array
array.append(more args)

this.execute()
```

Don't you think this looks cleaner

File: src/kubectl-object-model.ts [hunk 4]
```diff
@@ -9,10 +9,12 @@ export class Kubectl {
     private kubectlPath: string;
     private namespace: string;
     private ignoreSSLErrors: boolean;
+    private forceDeployment: boolean;
 
-    constructor(kubectlPath: string, namespace?: string, ignoreSSLErrors?: boolean) {
+    constructor(kubectlPath: string, namespace?: string, ignoreSSLErrors?: boolean, forceDeployment?: boolean) {
```
Comment (ds-ms) [inline src/kubectl-object-model.ts:14]: Also, given that this is only to be used for `apply` doesn't it make more sense to be a parameter to that function instead?

**Pull request closed without merging**